webui/react/src/pages/TrialDetails/MultiTrialDetailsHyperparameters.tsx

```diff
@@ -95,13 +116,13 @@ const MultiTrialDetailsHyperparameters: React.FC<Props> = ({
     );
 
     return () => canceler.abort();
-  }, [experiment.id, isSupported, settings.filters.metric]);
+  }, [experiment.id, isSupported, settings.metric]);
 
   // Set the default filter batch.
   useEffect(() => {
-    if (!batches || batches.length === 0 || settings.filters.batch !== 0) return;
-    updateSettings({ filters: { ...settings.filters, batch: batches.first() } });
-  }, [batches, settings.filters, updateSettings]);
+    if (!batches || batches.length === 0 || settings.batch !== 0) return;
+    updateSettings({ batch: batches.first() });
```

Comment:
**question:** do you know if `updateSettings` takes a full setting or partial deltas? I'm looking at the code and I think it's implemented incorrectly? I looks like it can take in partial but the check of whether it's changed or not is assuming it's a full settings object...
```
  const updateSettings = useCallback(
    (updates: Settings) => {
      stateOb.update((s) =>
        s.set(
          config.storagePath,
          isEqual(s.get(config.storagePath), updates)
            ? s.get(config.storagePath) ?? {}
            : { ...s.get(config.storagePath), ...updates },
        ),
      );
    },
    [config, stateOb],
  );
```

Reply:
My understanding is that it can take a delta, and from a cursory reading of that code I think that passing in a delta should work. At worst it seems like there might be a slight inefficiency in the case where you pass in a delta where the key/value pair matches the current state.

Reply:
FYI, [related PR](https://github.com/determined-ai/determined/pull/6894) that stemmed from this conversation :D

TL;DR - code-wise it was only supporting full settings but due to incorrect typings it allowed partials through.